src/parquet/util/rle-encoding.h

```diff
@@ -114,6 +115,10 @@ class RleDecoder {
   template <typename T>
   int GetBatch(T* values, int batch_size);
 
+  /// Like GetBatch but the values are then decoded using the provided dictionary
+  template <typename T>
+  int GetBatchWithDict(T* values, int batch_size, Vector<T>& dictionary);
```

Comment:
It may be better to put the dictionary as the first argument (https://google.github.io/styleguide/cppguide.html#Function_Parameter_Ordering)